python/vowpalwabbit/pyvw.py

```diff
@@ -883,8 +883,8 @@ def __init__(self, ex, id):
             if len(id) == 0:
                 id = " "
             self.id = None  # we don't know and we don't want to do the linear search required to find it
-            self.ns = id[0]
-            self.ord_ns = ord(self.ns)
+            self.ns = id
```

Comment:
Storing the hashed namespace here will allow it to be reused rather than hashed in every function call.